### ISSUE TITLE
NoSuchMethodError 'java.sql.Connection com.intellij.database.util.DbImplUtil.getConnection(com.intellij.database.psi.DbElement)'

### ISSUE THREAD

**Issue by eliashkan** [kTT/dbunit-extractor#21]
**Describe the bug**
Message from the title is displayed in a green text balloon when executing in place extraction in an xml file, but query remains unchanged.

**To Reproduce**
1. Create postgres database
2. Hit "convert query to XML" on SELECT * FROM PROCEDURE_TASK in an XML file

**Expected behavior**
Should convert to the existing records.

**Screenshots**
![Screenshot 2021-03-24 at 02 42 21 (2)](https://user-images.githubusercontent.com/38148675/112241545-bd84d600-8c4a-11eb-923c-8570f36cc417.png)

**Desktop (please complete the following information):**
 - OS: MacOS Catalina 10.15.7
 - Version of dbunit-extractor: Latest

**Notes**
- Copying records from the database view by selecting and right-clicking does work.

**Comment by eliashkan** [comment]
I found where the options for the plugin are and set the default datasource but now the popup is gone and it still does nothing when trying extract XML in place.

**Comment by eliashkan** [comment]
I found I might have to write my query like so: select * from matchitdb.public."PROCEDURE";
It still does nothing though.

**Comment by eliashkan** [comment]
```
java.lang.NoSuchMethodError: 'java.sql.Connection com.intellij.database.util.DbImplUtil.getConnection(com.intellij.database.psi.DbElement)'
	at li.ktt.QueryToXMLConverter.applySelectionChange(QueryToXMLConverter.java:139)
	at li.ktt.QueryToXMLConverter.access$000(QueryToXMLConverter.java:48)
	at li.ktt.QueryToXMLConverter$1.run(QueryToXMLConverter.java:106)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:268)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by eliashkan** [comment]
Does this have something to do with the intelliJ version, since it's not picking up a method you're using in QueryToXMLConverter that's in com.intellij ... ?

**Comment by eliashkan** [comment]
IntelliJ IDEA 2020.2.3 (Ultimate Edition)
Build #IU-202.7660.26, built on October 6, 2020
Runtime version: 11.0.8+10-b944.34 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 10.15.7
GC: ParNew, ConcurrentMarkSweep
Memory: 2014M
Cores: 4
Registry: compiler.automake.allow.when.app.running=true, ide.balloon.shadow.size=0, appcode.compiler.invocations.disk.cache=true
Non-Bundled Plugins: IdeaVIM, Lombook Plugin, manjaro.mpb, com.intellij.selenium, org.jetbrains.kotlin, li.ktt, izhangzhihao.rainbow.brackets, org.jetbrains.plugins.go

**Comment by eliashkan** [comment]
I have com.intellij.database but the util dir is missing. I don't think it's still bundled in the latest versions of IDEA. You might want to bundle them. 
https://youtrack.jetbrains.com/issue/DBE-9194

**Comment by eliashkan** [comment]
https://intellij-support.jetbrains.com/hc/en-us/community/posts/206121339-Can-I-access-and-use-the-IDEA-Data-Sources-

> 
> Avatar
> Gregory Shrago
> Created June 03, 2013 19:55
> Comment actions
> 
> In order to simplify this a
> 
> public DataSource.getConnection(Project)
> 
> method is added to IDEA 13 and IDEA 12.1.X codebases.
>

**Comment by eliashkan** [comment]
```
java.lang.NoClassDefFoundError: com/intellij/injected/editor/EditorWindowImpl
	at li.ktt.QueryToXMLConverter.isAvailable(QueryToXMLConverter.java:287)
	at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.availableFor(ShowIntentionActionsHandler.java:153)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.lambda$getActionsToShow$1(ShowIntentionsPass.java:336)
	at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.chooseBetweenHostAndInjected(ShowIntentionActionsHandler.java:190)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:335)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doCollectInformation(ShowIntentionsPass.java:240)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:400)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:393)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:392)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:368)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:167)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:178)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:366)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by eliashkan** [comment]
2021-04-16 15:02:31,215 [2057922]  ERROR - aemon.impl.PassExecutorService - com/intellij/injected/editor/EditorWindowImpl 
java.lang.NoClassDefFoundError: com/intellij/injected/editor/EditorWindowImpl
	at li.ktt.QueryToXMLConverter.isAvailable(QueryToXMLConverter.java:287)
	at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.availableFor(ShowIntentionActionsHandler.java:153)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.lambda$getActionsToShow$1(ShowIntentionsPass.java:336)
	at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.chooseBetweenHostAndInjected(ShowIntentionActionsHandler.java:190)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:335)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doCollectInformation(ShowIntentionsPass.java:240)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:400)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:393)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:392)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:368)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:167)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:178)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:366)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.ClassNotFoundException: com.intellij.injected.editor.EditorWindowImpl PluginClassLoader(plugin=PluginDescriptor(name=DbUnit Extractor, id=li.ktt, descriptorPath=plugin.xml, path=~\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\211.6693.111.plugins\dbunit-extractor.jar, version=1.5, package=null), packagePrefix=null, instanceId=195, state=active)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:215)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 24 more
2021-04-16 15:02:31,216 [2057923]  ERROR - aemon.impl.PassExecutorService - IntelliJ IDEA 2021.1  Build #IU-211.6693.111 
2021-04-16 15:02:31,221 [2057928]  ERROR - aemon.impl.PassExecutorService - JDK: 11.0.10; VM: Dynamic Code Evolution 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2021-04-16 15:02:31,222 [2057929]  ERROR - aemon.impl.PassExecutorService - OS: Windows 10 
2021-04-16 15:02:31,222 [2057929]  ERROR - aemon.impl.PassExecutorService - Plugin to blame: DbUnit Extractor version: 1.5 
2021-04-16 15:02:31,222 [2057929]  ERROR - aemon.impl.PassExecutorService - Last Action: Stop 
2021-04-16 15:02:31,224 [2057931]  ERROR - aemon.impl.PassExecutorService - com/intellij/injected/editor/EditorWindowImpl 
java.lang.NoClassDefFoundError: com/intellij/injected/editor/EditorWindowImpl

**Comment by kTT** [comment]
@eliashkan I've made fixes, please try with version 1.8 and let me know if problem still occurs.
